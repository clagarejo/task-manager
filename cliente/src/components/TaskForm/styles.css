.task-form {
    display: flex;
    flex-direction: column;
    gap: 15px;
    width: 100%;
    max-width: 600px;
    padding: 20px;
    margin-top: -25px;
}

.task-input,
.task-textarea {
    width: 100%;
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 4px;
    font-size: 1rem;
    box-sizing: border-box;
}

.task-textarea {
    resize: none;
    min-height: 100px;
}

.task-button {
    width: 100%;
    background-color: #007bff;
    color: white;
    border: none;
    border-radius: 4px;
    font-size: 1rem;
    cursor: pointer;
    transition: background-color 0.3s;
    padding: 15px;
}

.task-button:hover {
    background-color: #0056b3;
}

input,
textarea {
    font-family: Arial, sans-serif;
}

button:disabled {
    background-color: #ccc;
    cursor: not-allowed;
}