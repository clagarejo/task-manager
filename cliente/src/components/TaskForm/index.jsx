import React from 'react';
import './styles.css';

const TaskForm = ({
    handleAddOrUpdateTask,
    editingTask,
    newTitle,
    newDescription,
    setNewTitle,
    setNewDescription
}) => {
    const handleSubmit = (e) => {
        e.preventDefault();

        if (newTitle.trim() && newDescription.trim()) { // Asegúrate de que los campos no estén vacíos
            handleAddOrUpdateTask({ title: newTitle, description: newDescription });
        } 
    };

    return (
        <form onSubmit={handleSubmit} className="task-form">
            <input
                type="text"
                value={newTitle}
                onChange={(e) => setNewTitle(e.target.value)}
                placeholder="Título"
                className="task-input"
            />
            <textarea
                value={newDescription}
                onChange={(e) => setNewDescription(e.target.value)}
                placeholder="Descripción"
                className="task-textarea"
            />
            <button type="submit" className="task-button">
                {editingTask ? "Actualizar tarea" : "Agregar tarea"}
            </button>
        </form>
    );
};

export default TaskForm;
