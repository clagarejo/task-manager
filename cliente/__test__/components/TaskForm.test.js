import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import TaskForm from '../../src/components/TaskForm';

test('debe coincidir con el snapshot', () => {
    const { asFragment } = render(<TaskForm onAddTask={() => { }} />);

    // Verifica si el componente se renderiza de acuerdo al snapshot
    expect(asFragment()).toMatchSnapshot();
});

// test('debe llamar onAddTask cuando el formulario es válido', () => {
//     const onAddTaskMock = jest.fn();

//     render(<TaskForm onAddTask={onAddTaskMock} />);

//     // Completa los campos del formulario
//     fireEvent.change(screen.getByPlaceholderText(/Título/i), { target: { value: 'Nueva tarea' } });
//     fireEvent.change(screen.getByPlaceholderText(/Descripción/i), { target: { value: 'Descripción de la tarea' } });
    
//     // Simula el envío del formulario
//     fireEvent.click(screen.getByText(/agregar tarea/i));

//     // Verifica que la función onAddTask haya sido llamada con los valores correctos
//     expect(onAddTaskMock).toHaveBeenCalledWith({
//         title: 'Nueva tarea',
//         description: 'Descripción de la tarea',
//     });
// });

test('llama a handleAddOrUpdateTask con los valores correctos al enviar el formulario', () => {
    const mockHandleAddOrUpdateTask = jest.fn();

    render(
        <TaskForm
            handleAddOrUpdateTask={mockHandleAddOrUpdateTask}
            editingTask={null}
            newTitle="Nueva tarea"
            newDescription="Descripción de la nueva tarea"
            setNewTitle={() => { }}
            setNewDescription={() => { }}
        />
    );

    // Simula el envío del formulario
    fireEvent.submit(screen.getByRole('button', { name: /agregar tarea/i }));

    // Asegura que la función se llamó con los valores correctos
    expect(mockHandleAddOrUpdateTask).toHaveBeenCalledWith({
        title: 'Nueva tarea',
        description: 'Descripción de la nueva tarea',
    });
});

test('no llama a handleAddOrUpdateTask si los campos están vacíos', () => {
    const mockHandleAddOrUpdateTask = jest.fn();

    render(
        <TaskForm
            handleAddOrUpdateTask={mockHandleAddOrUpdateTask}
            editingTask={null}
            newTitle=""
            newDescription=""
            setNewTitle={jest.fn()}
            setNewDescription={jest.fn()}
        />
    );

    fireEvent.click(screen.getByRole('button'));

    // Verifica que handleAddOrUpdateTask no se haya llamado
    expect(mockHandleAddOrUpdateTask).not.toHaveBeenCalled();
});






