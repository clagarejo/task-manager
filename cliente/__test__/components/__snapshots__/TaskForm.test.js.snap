// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe coincidir con el snapshot 1`] = `
<DocumentFragment>
  <form
    class="task-form"
  >
    <input
      class="task-input"
      placeholder="Título"
      type="text"
      value=""
    />
    <textarea
      class="task-textarea"
      placeholder="Descripción"
    />
    <button
      class="task-button"
      type="submit"
    >
      Agregar tarea
    </button>
  </form>
</DocumentFragment>
`;
